Echo/EchoNode.cpp

```diff
@@ -136,25 +152,56 @@ std::vector< std::string > PolySyncEcho::getAvailableMessageNames()
     {
         messageNames.emplace_back( getMessageNameByType( index ) );
     }
+
     return messageNames;
 }
 
 
-void PolySyncEcho::printAvailableMessage( const std::vector< std::string > & messageTypeStrings )
+void PolySyncEcho::printAvailableMessage(
+        const std::vector< std::string > & messageTypeStrings ) const
 {
     for( auto messageTypeString : messageTypeStrings )
     {
-        cout << "    " << messageTypeString << endl;
+        std::cout << "    " << messageTypeString << std::endl;
     }
 }
 
 
-void PolySyncEcho::printEchoHelpPage()
+void PolySyncEcho::printEchoHelpPage() const
 {
     echoHelp.printEchoHelp();
 }
 
 
-// END polysync::PolySyncEcho class
+void PolySyncEcho::printActiveMessageTypes() const
+{
+    polysync::RuntimeInfo info( *this );
 
-} // END namespace polysync
+    auto typeInfoList = info.getDiscoveredMessageTypes();
+
+    std::cout << std::endl << std::endl
+              << "Active PolySync Message Types: "
+              << std::endl;
+
+    for( auto typeInfo : typeInfoList )
+    {
+        if( typeInfo.getMessageType() != PSYNC_MSG_TYPE_INVALID )
+        {
+            std::cout << "    "
+                      << typeInfo.getTypeName()
+                      << std::endl;
+        }
+    }
+}
+
+
+bool PolySyncEcho::durationReached() const
+{
+    return ( polysync::getTimestamp() - _applicationStartTime ) >= getDuration();
+}
+
+
+ps_timestamp PolySyncEcho::getDuration() const
+{
+    return _inputHandler.getUserSpecifiedDuration() * 1e6;
```

Comment:
Magic number.  Might want to create an internal const (like OneSecond ) for doing this conversion.